### DISCUSSION TITLE
In-Depth explanation of how to assign a Type Property in xaml

### DISCUSSION THREAD

**Discussion by Sarund9** [AvaloniaUI/Avalonia#5701] [Q&A] [answered]
I am making a TreeView, and when creating the DataTemplate i have to assign the DataType Property.
But i keep getting " Unable to Resolve Type from namespace  https://github.com/avaloniaui " error.

Code:
```cs
namespace MyAvaloniaApp
{
    public interface IMyInterface
    {
        string Name { get; set; }
    }
}
```
XAML:
```xaml
<TreeDataTemplate DataType="MyAvaloniaApp.IMyInterface">
          
        </TreeDataTemplate>
```
Could someone give an explanation of how to assign a Type in xaml, or link to documentation about this (I couldn't find any)

**Comment by maxkatz6** [comment, marked as answer]
You need to define namespace:
```xaml
<TreeDataTemplate xmlns:namespace="using:MyAvaloniaApp"
                                DataType="namespace:IMyInterface">
          
</TreeDataTemplate>
```
Or WPF style:
```xaml
<TreeDataTemplate xmlns:namespace="clr-namespace:MyAvaloniaApp;assembly=MyAssemblyName"
                                DataType="namespace:IMyInterface">
          
</TreeDataTemplate>
```

This "xmlns" part is usually set on the root element of xml/xaml file.
